src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php

```diff
@@ -0,0 +1,395 @@
+                $response
+            );
+        }
+
+        dump(sprintf('[Crowdin info] File "%s.%s" updated successfully.', $domain, 'xlf'));
+    }
+
+    private function uploadTranslations(int $fileId, string $domain, string $content, string $locale): void
+    {
+        if (!$fileId) {
+            return;
+        }
+
+        $storageId = $this->addStorage($domain, $content);
+
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.translations.postOnLanguage (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.translations.postOnLanguage (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/projects/%s/translations/%s', $this->getEndpoint(), $this->projectId, $locale), [
+            'headers' => array_merge($this->getDefaultHeaders(), [
+                'Content-Type' => 'application/json',
+            ]),
+            'body' => json_encode([
+                'storageId' => $storageId,
+                'fileId' => $fileId,
+            ]),
+        ]);
+
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new ProviderException(
+                sprintf('Unable to upload translations to Crowdin for domain "%s" and locale "%s".', $domain, $locale),
+                $response
+            );
+        }
+
+        dump(sprintf('[Crowdin info] "%s" translations for "%s.%s" file uploaded successfully.', $locale, $domain, 'xlf'));
+    }
+
+    private function exportProjectTranslations(string $languageId, int $fileId): string
+    {
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.translations.exports.post (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.translations.exports.post (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/projects/%d/translations/exports', $this->getEndpoint(), $this->projectId), [
+            'headers' => array_merge($this->getDefaultHeaders(), [
+                'Content-Type' => 'application/json',
+            ]),
+            'body' => json_encode([
+                'targetLanguageId' => $languageId,
+                'fileIds' => [$fileId],
+            ]),
+        ]);
+
+        if (Response::HTTP_NO_CONTENT === $response->getStatusCode()) {
+            throw new ProviderException(
+                sprintf('No content in exported file %d.', $fileId),
+                $response
+            );
+        }
+
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new ProviderException(
+                sprintf('Unable to export file %d translations for language %s.', $fileId, $languageId),
+                $response
+            );
+        }
+
+        $export = json_decode($response->getContent(), true);
+
+        return file_get_contents($export['data']['url']);
```

Comment:
You must use the HttpClient to fetch the content of the export file here, to handle potential failure properly.